graylog2-web-interface/src/components/bootstrap/BootstrapModalForm.jsx

```diff
@@ -18,12 +18,12 @@ class BootstrapModalForm extends Component {
     this.open = this.open.bind(this);
     this.close = this.close.bind(this);
     this._submit = this._submit.bind(this);
-    this._onModalClose = this._onModalClose.bind(this);
+    this._onModalCancel = this._onModalCancel.bind(this);
   }
 
-  _onModalClose() {
-    if (typeof this.props.onModalClose === 'function') {
-      this.props.onModalClose();
+  _onModalCancel() {
+    if (typeof this.props.onCancel === 'function') {
```

Comment:
The MTU case implies that there is some drafting mechanism, which we do not have (and is probably overkill for our use case). Those applications which are much closer to our use case are all following the semantics implemented by this PR, which means that when the user is cancelling (the action of doing this is btw a user input as well) the "draft" is discarded.

Another argument for changing the behavior we have today: when editing an existing input, making changes in one of the fields, cancelling the form and editing the entity again shows the changes that were just done and cancelled. This is imo unexpected for a majority of the users and leads to an inconsistency between the persisted state of the entity, the state that is shown in the overview and the state as shown in the edit form.